sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -140,6 +166,7 @@ object SQLExecution {
     } finally {
       executionIdToQueryExecution.remove(executionId)
       sc.setLocalProperty(EXECUTION_ID_KEY, oldExecutionId)
+      unsetRootExecutionId(sc, oldExecutionId)
```

Comment:
should it be `unsetRootExecutionId(sc, executionId)` ?

Reply:
done